python/cudf/cudf/_lib/pylibcudf/column.pyx

```diff
@@ -296,3 +332,29 @@ cdef class ListColumnView:
     cpdef offsets(self):
         """The offsets column of the underlying list column."""
         return self._column.child(1)
+
+
+def _data_type_from_iface(iface):
```

Comment:
*question*: Does it make sense to cache the results, so:
```
@functools.cache
def datatype_from_dtype_desc(desc):
   mapping = {...}
   return DataType(mapping[desc])

def datatype_from_iface(iface):
    return datatype_from_dtype_desc(iface["typestr"][1:])
```

Reply:
Done